src/backend.c

```diff
@@ -503,29 +512,30 @@ static bool write_kernel_binary (hashcat_ctx_t *hashcat_ctx, char *kernel_file,
 {
   if (binary_size > 0)
   {
-    FILE *fp = fopen (kernel_file, "wb");
+    HCFILE fp;
 
-    if (fp == NULL)
+    // change HCFILE_FORMAT_GZIP to HCFILE_FORMAT_PLAIN to write kernel binary uncompressed
+    if (hc_fopen (&fp, kernel_file, "wb", HCFILE_FORMAT_GZIP) == false)
```

Comment:
Good thing is, after removing writing files compressed, you can also remove third parameter for hc_fopen() which kind of defeats a normal wrapper functionality.

Reply:
yep